test/performance/kshelper.py

```diff
@@ -106,72 +111,138 @@ def get_crd_time(self, grp, v, pl, name=None, check=False):
-                    fout.write(name + "\t" + creation_time + "\t" + update_time + "\t" + status_condition + "\t" + op_manager + "\n")
-            fout.close()
+                            continue
+
+                    output=obj_name + "\t" + creation_timestamp + "\t" + update_timestamp + "\t" + status_condition + "\t" + op_manager
+                    if name != None and single == True:
+                       return output
+                    else:
+                       fout.write(output+ "\n")
+
+            if name == None or single == False:
+               fout.flush()
+               fout.close()
+            return None
+
         except ApiException as e:
             print("Exception when calling client->CustomObjectsApi(): %s\n" % e)
 
 
-    def get_pod_time(self, ns, name=None):
+    def get_obj_time(self, fname, kind, ns, name=None):
         try:
+            kind = kind.lower()
             if name != None:
-                api_response = self.core_v1_api.list_namespaced_pod(ns, field_selector='metadata.name=' + name)
-            else: 
-                api_response = self.core_v1_api.list_namespaced_pod(ns)
+                if kind == "pod" or kind == "pods":
+                   api_response = self.core_v1_api.read_namespaced_pod(name, ns)
+                elif kind == "deployment" or kind == "deployments":
+                   api_response = self.app_v1_api.read_namespaced_deployment(name, ns)
+                elif kind == "secret" or kind == "secrets":
+                   api_response = self.core_v1_api.read_namespaced_secret(name, ns)
+                elif kind == "configmap" or kind == "configmaps":
+                   api_response = self.core_v1_api.read_namespaced_config_map(name, ns)
+                elif kind == "service" or kind == "services":
+                   api_response = self.core_v1_api.read_namespaced_service(name, ns)
+                else:
+                    print("Not supported object kind ...")
+                    return
+            else:
+                if kind == "pod" or kind == "pods": 
+                   api_response = self.core_v1_api.list_namespaced_pod(ns)
+                elif kind == "deployment" or kind == "deployments":
+                   api_response = self.app_v1_api.list_namespaced_deployment(ns)
+                elif kind == "secret" or kind == "secrets":
+                   api_response = self.core_v1_api.list_namespaced_secret(ns)
+                elif kind == "configmap" or kind == "configmaps":
+                   api_response = self.core_v1_api.list_namespaced_config_map(ns)
+                elif kind == "service" or kind == "services":
+                   api_response = self.core_v1_api.list_namespaced_service(ns)
+                else:
+                    print("Not supported object kind ...")
+                    return
+                
+                # Create the output file
+                fout = open(fname, 'a')
 
             object_list = api_response.items
```

Comment:
If this right if `name != None` ?

Reply:
Yes, that's still correct. This condition has been tested.

Reply:
Fixed!